boards/shields/amg88xx/amg88xx_grid_eye_eval_shield.overlay

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) 2022 Panasonic Industrial Devices Europe GmbH
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+&i2c0_default {
+	group1 {
+		bias-pull-up;
+	};
+};
```

Comment:
`i2c0_default` is vendor specific. Using this defeats the purpose of defining arduino generic shield
One solution is to provide boards specific overlays as done on other shields `boards/xxx/xxx.overlay`.

Reply:
Okay. But this means that I could add one overlay file for every board in the current ``platform_allow`` list to ``boards/xxx/`` folder, that having exactly the same content right? Just the file name is different. 

And I'm getting a little bit confused when having an board specific overlay in a boards folder, which is just valid for one of the two available shields in folder ``zephyr\boards\shields\amg88xx``? How can a user distinguish? Or is it necessary, to have that new shield on top level like ``zephyr\boards\shields\amg88xx_v2`` and leave the old ``amg88xx`` shield folder untouched?

Reply:
Then why do not you fix it in your board dts?

Reply:
Because these pull-up's on i2c lines are specific for this shield. If I add this to the evaluation boards devicetree they are set for all possible sensors that can be connected in the future and that's maybe not wanted or can trigger some unwanted behavior.

Reply:
> Because these pull-up's on i2c lines are specific for this shield.

Why?

> If I add this to the evaluation boards devicetree they are set for all possible sensors that can be connected in the future and that's maybe not wanted or can trigger some unwanted behavior.

What are differences if this shield is following you current approach and is combined with "all possible sensors that can be connected in the future"?

Reply:
I try to explain my understanding a little bit more in detail. Sorry, I'm still new to zephyr and trying to understand how everything is working together.

The result should be, that the shield is working on as many boards as possible, right. 

> Then why do not you fix it in your board dts?

This wouldn't be enough, if I'm getting it right? With this approach i have to add the ``bias-pull-up`` to every single i2c ``*-pinctrl.dtsi`` file of the ~300 available boards in ``zephyr/boards/arm``? Otherwise it could be, that you get the same error on i2c line I've described above?

So in my understanding, it has to be the shield that enables the pull-up's, because they are just needed in this setup for this shield. If I'm also add the pull-up's to other boards in ``zephyr/boards/arm``, they have this pull-up's always activated for other setups, with other shields and other sensors. And that feels somehow wrong for me.


So in my opinion there are now 3 different ways to fix this:

1. Use my current approach, described in https://github.com/zephyrproject-rtos/zephyr/issues/53175 by gmarull, and have vendor specific i2c0_default as an pin configuration state node in shield overlay. This leads to a platform_allow list in ``samples/sensor/amg88xx/sample.yaml`` to exclude all boards that don't have this vendor specific i2c0_default node.

2. Going with the approach of @erwango and add a lot of board overlays in new folder ``shields\amg88xx\boards``. So i would be able to add the pull-up's for the correct boards, dependent on the naming of the i2c node in ``*-pinctrl.dtsi`` file. But his also means, that i will have a lot of overlays here, that having the same content, just other names. And I guess this approach would also cause the necessity to add the shield as an separate shield on top level (e.g. ``zephyr\boards\shields\amg88xx_v2``) because it's difficult to distinguish between old ``amg88xx_eval_kit`` and new ``amg88xx_grid_eye_eval_shield`` otherwise.

3. Use your approach above and change all the boards ``*-pinctrl.dtsi`` files directly, with the pro, that i don't have to use a platform_allow list anymore and can remove the bias-pull-up from my shield overlay, but the cons described above.

For me it looks like none of this three options are fully satisfying. What would you say? Which option would you implement now?

Reply:
Either I²C pull-ups are present on the board or it is configured as a property of GPIO, if possible. The presence of a physical pull-up on the shield makes combination of the shields difficult. Finding out which boards need pull-up and adding it as an overlay for those boards to a shield is total nonsense.

Reply:
You are right. I will remove that part. And will describe, that a user simply has to add the jumpers on the shield, if he really need pull up's.